icu4c/source/test/cintltst/udatpg_test.c

```diff
@@ -836,4 +840,46 @@ static void TestRegionOverride(void) {
         udatpg_close(dtpg);
     }
 }
+
+static void TestISO8601(void) {
+    typedef struct TestCase {
+        const char* locale;
+        const UChar* skeleton;
+        const UChar* expectedPattern;
+    } TestCase;
+
+    const TestCase testCases[] = {
+        { "en_GB@calendar=iso8601;rg=uszzzz", u"EEEEyMMMMdjmm", u"y MMMM d, EEEE 'at' h:mm a" },
+        { "en_GB@calendar=iso8601;rg=uszzzz", u"EEEEyMMMMdHmm", u"y MMMM d, EEEE 'at' HH:mm" },
+        { "en_GB@calendar=iso8601;rg=uszzzz", u"Edjmm",         u"d, EEE, h:mm a" },
+        { "en_GB@calendar=iso8601;rg=uszzzz", u"EdHmm",         u"d, EEE, HH:mm" },
+
+        { "en_US@calendar=iso8601",           u"EEEEyMMMMdjmm", u"y MMMM d, EEEE 'at' h:mm a" },
+        { "en_US@calendar=iso8601",           u"EEEEyMMMMdHmm", u"y MMMM d, EEEE 'at' HH:mm" },
+        { "en_US@calendar=iso8601",           u"Edjmm",         u"d, EEE, h:mm a" },
+        { "en_US@calendar=iso8601",           u"EdHmm",         u"d, EEE, HH:mm" },
+
+        { "en_US",                            u"EEEEyMMMMdjmm", u"EEEE, MMMM d, y 'at' h:mm a" },
+        { "en_US",                            u"EEEEyMMMMdHmm", u"EEEE, MMMM d, y 'at' HH:mm" },
+        { "en_US",                            u"Edjmm",         u"d EEE, h:mm a" },
+        { "en_US",                            u"EdHmm",         u"d EEE, HH:mm" },
```

Comment:
@macchiati Regular US English wants `d EEE`?  "12 Tuesday"?  I don't think I've ever heard it expressed that way...

Reply:
We decided on ordering y m d dow h m s z

We expect to be refining the patterns in later iterations. So this is just capturing what CLDR does now.

Reply:
BTW ISO 8601 doesn't allow for the dow feature

Reply:
This one isn't ISO 8601, though.  It's just generic US English (for comparison).  So is your comment still operative?